### PR TITLE
chore: Ensure privilege validations for API endpoints

### PR DESCRIPTION
**Pull request by sojan-official**
- Add necessary authorizations for agents and admins for various API endpoints

# Testing 

- Permission differences between agents and  administrators 
- Agents shouldn't be able to access conversations outside inboxes they have access to ( via direct URLs)
- Agents should be able to interact with crud APIs for inbox member additions 
- Contactable Inboxes on the contacts page should only list inboxes to which the agent has access to

### CODE REVIEW

File: app/controllers/api/v1/accounts/contacts/conversations_controller.rb [hunk 1]
```diff
@@ -8,9 +8,7 @@ def index
   private
 
   def inbox_ids
-    if Current.user.administrator?
-      Current.account.inboxes.pluck(:id)
-    elsif Current.user.agent?
```
Comment (pranavrajs) [inline app/controllers/api/v1/accounts/contacts/conversations_controller.rb:13]: If the user is an admin, I guess we could show every inbox available.

Comment (sojan-official) [reply app/controllers/api/v1/accounts/contacts/conversations_controller.rb:13]: `assigned_inboxes` handle this condition. it returns all inboxes for admin users and only the specific inboxes for agents

**Comment by pranavrajs** [comment]
@sojan-official Looks like the new APIs have an issue with access. Otherwise, it looks fine, let us get it merged soon.

**Comment by sojan-official** [comment]
@pranavrajs  back to you

**Review by pranavrajs — APPROVED** [review]

**Pull request merged**